### PR TITLE
doc-parser 

### PR DESCRIPTION
**Pull request by ScottSuarez**
add tool for translating google cloud docs to yaml
**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/tools/doc-parser/index.mjs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    fields = window.document.querySelectorAll(`tr[id^=${object}\\.FIELDS]`)
+  }
+  fields = [... fields]
+  fields = fields.filter( element => shouldBeFiltered(element.getAttribute("id")))
+  var out = await Promise.all(fields.map( async element => {
+    console.log(element.getAttribute("id"))
+    var row = [... element.querySelectorAll(`td`)]
+    var name = row[0].textContent
+    if (row[1] == undefined) {
+      console.log("hi scott")
```
Comment (slevenick) [inline mmv1/tools/doc-parser/index.mjs:86]: Should most of these console.log lines be removed?

File: mmv1/tools/doc-parser/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,1082 @@
+{
```
Comment (slevenick) [inline mmv1/tools/doc-parser/package-lock.json:1]: I'm not super familiar with node development, do we need to commit the lock file?

File: mmv1/tools/doc-parser/index.mjs (hunk 1 above)
Comment (ScottSuarez) [reply mmv1/tools/doc-parser/index.mjs]: I removed the hi, scott log. The other ones are useful to see the serialization process in action and that things are happening. I could hide it behind a debug flag or just leave it? wdyt?

File: mmv1/tools/doc-parser/package-lock.json (hunk 2 above)
Comment (ScottSuarez) [reply mmv1/tools/doc-parser/package-lock.json:1]: yeah, lock is tell what dependencies to use. It ensure if someone runs the tool they are using the same versions. Similar to `go.sum`

**Review by slevenick — APPROVED** [review]
How much do you trust the output of this tool?

My hesitation to use similar generators that read the discovery docs has always been that different products have different ways of representing the same things. One may add `Required.` to the descriptions of required fields, while another won't. I'd be careful using the output of this, and be sure to test whatever comes out!

That said, this could be useful to get a place to start!

**Comment by ScottSuarez** [comment]
> How much do you trust the output of this tool?
> 
> My hesitation to use similar generators that read the discovery docs has always been that different products have different ways of representing the same things. One may add `Required.` to the descriptions of required fields, while another won't. I'd be careful using the output of this, and be sure to test whatever comes out!
> 
> That said, this could be useful to get a place to start!

It's definitely not perfect. I think, for example it can't really handle exclusion scenarios. ie I need either x, or y. 

I'll document some of the limitations in the readme

**Pull request merged**